File: examples/Training/python/Parse_OMERO_Properties.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

#
# Copyright (C) 2015 University of Dundee & Open Microscopy Environment.
#                    All Rights Reserved.
# Use is subject to license terms supplied in LICENSE.txt
#

"""
FOR TRAINING PURPOSES ONLY!
"""

import omero

client = omero.client()

omeroProperties = client.getProperties().getPropertiesForPrefix('omero')

# Set encoding so that print statements or str(unicode) don't fail
# in testing. See https://github.com/openmicroscopy/openmicroscopy/pull/5400
import sys
reload(sys)
sys.setdefaultencoding('utf-8')

# Configuration
# =================================================================
# These values will be imported by all the other training scripts.
HOST = omeroProperties.get('omero.host', 'localhost')
PORT = omeroProperties.get('omero.port', 4064)
USERNAME = omeroProperties.get('omero.user')
PASSWORD = omeroProperties.get('omero.pass')
OMERO_WEB_HOST = omeroProperties.get('omero.webhost')
SERVER_NAME = omeroProperties.get('omero.servername')
projectId = omeroProperties.get('omero.projectid')
datasetId = omeroProperties.get('omero.datasetid')
imageId = omeroProperties.get('omero.imageid')
plateId = omeroProperties.get('omero.plateid')

File: examples/Training/matlab/parseOmeroProperties.m
% Copyright (C) 2014 University of Dundee & Open Microscopy Environment.
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; either version 2 of the License, or
% (at your option) any later version.
%
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
%
% You should have received a copy of the GNU General Public License along
% with this program; if not, write to the Free Software Foundation, Inc.,
% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

function props = parseOmeroProperties(client)

omeroProperties = client.getProperties().getPropertiesForPrefix('omero');
props.hostname = char(omeroProperties.get('omero.host'));
props.username = char(omeroProperties.get('omero.user'));
props.password = char(omeroProperties.get('omero.pass'));
props.port = str2double(omeroProperties.get('omero.port'));
props.imageid = str2double(omeroProperties.get('omero.imageid'));
props.datasetid = str2double(omeroProperties.get('omero.datasetid'));
props.plateid = str2double(omeroProperties.get('omero.plateid'));
props.projectid = str2double(omeroProperties.get('omero.projectid'));
props.screenid = str2double(omeroProperties.get('omero.screenid'));
